File: app/src/main/java/com/coinninja/coinkeeper/model/db/converter/IdentityTypeConverter.java
package com.coinninja.coinkeeper.model.db.converter;

import com.coinninja.coinkeeper.model.db.enums.IdentityType;

import org.greenrobot.greendao.converter.PropertyConverter;

public class IdentityTypeConverter implements PropertyConverter<IdentityType, Integer> {

    @Override
    public IdentityType convertToEntityProperty(Integer databaseValue) {
        if (databaseValue == null) {
            return IdentityType.UNKNOWN;
        }

        for (IdentityType type : IdentityType.values()) {
            if (type.getId() == databaseValue) {
                return type;
            }
        }

        return IdentityType.UNKNOWN;
    }

    @Override
    public Integer convertToDatabaseValue(IdentityType type) {
        return type == null ? null : type.getId();
    }
}

File: persistance/src/main/java/app/coinninja/cn/persistance/converter/IdentityTypeConverter.kt
package app.coinninja.cn.persistance.converter

import androidx.room.TypeConverter
import app.coinninja.cn.persistance.model.IdentityType

object IdentityTypeConverter {

    @TypeConverter
    @JvmStatic
    fun to(id: Int?): IdentityType? = id?.let { return IdentityType.from(it) }

    @TypeConverter
    @JvmStatic
    fun from(type: IdentityType?): Int? = type?.let { return it.id }

}
